### ISSUE TITLE
Screen reader announces 'value' prop instead of 'title' or 'aria-label' prop

### ISSUE THREAD

**Issue by benbenbuhben** [react-component/tree-select#437]
**Problem:** 

The `value` prop of a tree node is often a numeric id, unique hash, or other non-human-readable identifier. Currently when navigating the Tree Select dropdown using a screen reader, each option's `value` prop is announced instead of either the `title` or `aria-label` prop. This makes it impossible for a screen reader user to understand the options being presented.

**Expected behavior:**

I would suggest that the screen reader prioritize which prop to announce in the following order:

1. `aria-label` prop: If provided, this should be the primary prop announced by the screen reader. It is expected to contain the most accessible and human-readable description of the node.
2. `title` prop: If the `aria-label` prop is not provided, the `title` prop should be announced since this is the actual content displayed in the dropdown. However, as the 'title' prop can be either a string or a React element, the screen reader should only announce it if it is of type string.
3. `value` prop: If neither an 'aria-label' prop nor a suitable 'title' prop (of string type) is provided, the `value` prop should be announced.

**Suspected Cause:**

The span element within the dropdown in OptionList.tsx, which has `aria-live="assertive"` applied, seems to always announce the `value` of the active node, even when an `aria-label` is present.

https://github.com/react-component/tree-select/blob/master/src/OptionList.tsx#L213-L215

**Additional Info:**

ant-design/ant-design#40055 - Existing Ant Design issue related to this problem.

ant-design/ant-design#41978 - Allows aria-label prop to be passed into rc-tree-select (released in antd@5.4.6)

react-component/tree-select#413 - Proposes a "labelInAriaLive" boolean prop, which if true would announce the `title` prop instead of the `value` prop. Although this does not account for an `aria-label` prop being set on the node, which should be prioritized by the screenreader first.

react-component/tree-select#426 - This PR attempted to solve this issue but was closed without being merged.
